src/Components/test/E2ETest/ServerRenderingTests/ErrorHandlingTest.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using TestServer;
+using Components.TestServer.RazorComponents;
+using Microsoft.AspNetCore.E2ETesting;
+using Xunit.Abstractions;
+using OpenQA.Selenium;
+
+namespace Microsoft.AspNetCore.Components.E2ETests.ServerRenderingTests;
+
+public class ErrorHandlingTest(BrowserFixture browserFixture, BasicTestAppServerSiteFixture<RazorComponentEndpointsStartup<App>> serverFixture, ITestOutputHelper output)
+    : ServerTestBase<BasicTestAppServerSiteFixture<RazorComponentEndpointsStartup<App>>>(browserFixture, serverFixture, output)
+{
+
+    [Fact]
+    public void RendersExceptionFromComponent()
+    {
+        GoTo("Throws");
+
+        Browser.Equal("Error", () => Browser.Title);
+
+        Assert.Collection(
+            Browser.FindElements(By.CssSelector(".text-danger")),
+            item => Assert.Equal("Error.", item.Text),
+            item => Assert.Equal("An error occurred while processing your request.", item.Text));
+        Browser.Equal("False", () => Browser.FindElement(By.Id("is-interactive-server")).Text);
```

Comment:
Do we somehow know for sure that this assertion won't pass just because we haven't yet activated the circuit, but will later go on to activate the circuit? Perhaps I'm missing something.

Reply:
Yes, this will activate the circuit.

I can make the test do the same thing that others do, where we call Blazor Start manually, and then revalidate that it's not replaced.

Reply:
OK, maybe the best we can do in this test is to wait a few seconds before asserting that the page content didn't change. It's quite hard to prove negatives :)

Reply:
I've updated the test to start blazor manually and validate that nothing becomes interactive.

Reply:
Don't you still need some kind of delay before the assertion on line 37? Just because the circuit has started doesn't mean that its first render has reached the browser yet, does it?

Reply:
I've added a 3s delay. Once we have initializers for blazor web, we should swich to use that.

Reply:
Do initializers have a callback to tell you when the first render has been applied to the DOM? I didn't think they did.